plivo.py

```diff
@@ -20,15 +22,28 @@ class PlivoError(Exception):
     pass
 
 
-def validate_signature(uri, post_params, signature, auth_token):
-    for k, v in sorted(post_params.items()):
+def validate_signature(uri, params, signature, auth_token):
+    qs = urlparse(uri).query
+    if qs:
+        params_from_qs = dict(parse_qsl(qs))
+        params_from_qs.update(params)
+        params = params_from_qs.copy()
+    s = uri.encode('utf-8')
+    for k, v in sorted(params.items()):
+        k = k.encode('utf-8')
         uri += k + v
+        if v is None:
+            x = ''
+        elif isinstance(v, basestring):
+            x = v.encode('utf-8')
+        else:
+            x = str(v)
+        params[k] = x
```

Comment:
Why would we override params? It's not expected that this function is going to alter any of the arguments we pass to it.